src/ray/raylet/node_manager.cc

```diff
@@ -470,28 +491,22 @@ void NodeManager::ResourceCreateUpdated(const ClientTableData &client_data) {
   return;
 }
 
-void NodeManager::ResourceDeleted(const ClientTableData &client_data) {
-  const ClientID client_id = ClientID::FromBinary(client_data.client_id());
+void NodeManager::ResourceDeleted(const ClientID &client_id,
+                                  const std::vector<std::string> &resource_names) {
   const ClientID &local_client_id = gcs_client_->client_table().GetLocalClientId();
 
-  ResourceSet new_res_set(
-      rpc::VectorFromProtobuf(client_data.resources_total_label()),
-      rpc::VectorFromProtobuf(client_data.resources_total_capacity()));
+  std::ostringstream oss;
+  for (auto &resource_name : resource_names) {
+    oss << resource_name << ", ";
+  }
```

Comment:
we should avoid doing this when debug log isn't enabled

Reply:
Do we have a way to detect if debug log is enabled?

Reply:
I found [this](https://github.com/ray-project/ray/blob/e55c8ca16594eb946d61169392c088cd7ea75adb/src/ray/util/logging.h#L14). 
But better wrap this in a macro (e.g., `RAY_LOG_ENABLED(DEBUG)`) to make the name more consistent with `RAY_LOG`.

Reply:
Done. Added macro `RAY_LOG_ENABLED`.